go/test/endtoend/vtgate/queries/orderby/orderby_test.go

```diff
@@ -83,3 +83,73 @@ func TestOrderBy(t *testing.T) {
 		mcmp.AssertMatches("select id1, id2 from t4 order by reverse(id2) desc", `[[INT64(5) VARCHAR("test")] [INT64(8) VARCHAR("F")] [INT64(7) VARCHAR("e")] [INT64(6) VARCHAR("d")] [INT64(2) VARCHAR("Abc")] [INT64(4) VARCHAR("c")] [INT64(3) VARCHAR("b")] [INT64(1) VARCHAR("a")]]`)
 	}
 }
+
+func TestOrderByComplex(t *testing.T) {
+	// tests written to try to trick the ORDER BY engine and planner
+	utils.SkipIfBinaryIsBelowVersion(t, 20, "vtgate")
```

Comment:
Should we change this once back ported? Since I think we need to back port this to v19 too since that has the previous PR and otherwise we have a significant regression?